src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -810,6 +810,15 @@ protected override ParseOptions OptionsCore
             CancellationToken cancellationToken)
             => ParseText(text, options, path, diagnosticOptions: null, cancellationToken);
 
+        // 2.8 BACK COMPAT OVERLOAD -- DO NOT MODIFY
+        public static SyntaxTree ParseText(
+            string text,
+            CSharpParseOptions options,
+            string path,
+            Encoding encoding,
+            CancellationToken cancellationToken)
+            => ParseText(text, options, path, encoding, diagnosticOptions: null, cancellationToken);
```

Comment:
i'm curious how this changed without being flagged in the first place.

Reply:
Our public API checker isn't smart enough for this. Adding a new optional parameter means means you have to manually update the public API file to:

1. Remove the old signature with the old optional parameters.
2. Add the new signature with the new optional parameters.
3. Add the new back compat signature.

In this case, 3 was missed in the original work.

Reply:
Ah ok, thanks!